### PR TITLE
Add Film and User models, in-memory storage, controllers, validation, and tests

### PR DESCRIPTION
**Pull request by AlpoAin**
- Create Film and User data classes with Jakarta Validation annotations
- Implement InMemoryFilmStorage and InMemoryUserStorage with auto-increment IDs and validation
- Add FilmController and UserController REST endpoints (POST, PUT, GET)
- Introduce ValidationException and GlobalExceptionHandler for error handling
- Add unit tests for Film and User validation edge cases

### CODE REVIEW

File: src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        log.info("Обновлён фильм: id={}", film.getId());
+        return film;
+    }
+
+    @Override
+    public Collection<Film> getAll() {
+        return new ArrayList<>(films.values());
+    }
+
+    private void validate(Film f) {
```
Comment (jeggga) [inline src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java]: Лучше вынести в отдельный утилитарный класс, где будут все проверки

File: src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        log.info("Обновлён пользователь: id={}", user.getId());
+        return user;
+    }
+
+    @Override
+    public Collection<User> getAll() {
+        return new ArrayList<>(users.values());
+    }
+
+    private void validate(User u) {
```
Comment (jeggga) [inline src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java]: Аналогично
